libraries/AP_AdvancedFailsafe/AP_AdvancedFailsafe.cpp

```diff
@@ -193,6 +200,9 @@ AP_AdvancedFailsafe::check(uint32_t last_heartbeat_ms, bool geofence_breached, u
     bool gcs_link_ok = ((now - last_heartbeat_ms) < 10000);
     bool gps_lock_ok = ((now - gps.last_fix_time_ms()) < 3000);
 
+    bool terminate_delay_comms = ((now - last_heartbeat_ms) > ((_terminal_delay * 1000) + 10000);
+    bool terminate_delay_gps   = ((now - gps.last_fix_time_ms()) > ((_terminal_delay * 1000) + 3000);
```

Comment:
If the delay is zero then both of these are always true after loss of link/lock, which means immediate termination - that's not the current behavior, which has no termination.

Reply:
I added a statement to set the flags to false if TERM_DLY is 0 or negative.  I still need to do some testing, but let me know if that should work.